app/src/main/java/com/example/traders/presentation/watchlist/WatchListRepository.kt

```diff
@@ -1,166 +1,167 @@
+            SortOrder.BY_NAME_ASC -> emitSortedByNameAsc()
+            SortOrder.BY_NAME_DESC -> emitSortedByNameDesc()
+            SortOrder.BY_CHANGE_ASC -> emitSortedByChangeAsc()
+            SortOrder.BY_CHANGE_DESC -> emitSortedByChangeDesc()
+        }
+    }
+
+    fun emitDefaultList() {
+        val cryptoListOrder = FixedCryptoList::class.enumConstantNames().toMutableList()
+        val orderBySymbol = cryptoListOrder.withIndex().associate { it.value to it.index }
+        _binanceCryptoList.update { it.sortedBy { orderBySymbol[it.symbol] } }
+    }
+
+    fun emitSortedByNameAsc() {
+        _binanceCryptoList.update { list ->  list.sortedBy { it.symbol } }
+    }
+
+    private fun emitSortedByNameDesc() {
+        _binanceCryptoList.update { list ->  list.sortedByDescending { it.symbol } }
+    }
+
+    private fun emitSortedByChangeAsc() {
+        _binanceCryptoList.update { list ->  list.sortedBy { it.priceChangePercent.toBigDecimal() } }
+    }
+
+    private fun emitSortedByChangeDesc() {
+        _binanceCryptoList.update { list ->  list.sortedByDescending { it.priceChangePercent.toBigDecimal() } }
+    }
+
+    suspend fun saveSortOrderOnPreference(sortOrder: SortOrder) {
+        preferencesManager.updateSortOrder(sortOrder)
+    }
+
+    suspend fun saveIsFavouriteOnPreference(isFavouritesOn: Boolean) {
+        preferencesManager.updateIsFavourite(isFavouritesOn)
+    }
+
+    suspend fun isFavouritesOn(): Boolean {
+        return preferencesManager.preferencesFlow.first().isFavourite
+    }
+
+    suspend fun sortOrderType(): SortOrder {
+        return preferencesManager.preferencesFlow.first().sortOrder
+    }
+
+    private fun List<Binance24DataItem>.getFixedCryptoList(): List<Binance24DataItem> {
+        return filter { element ->
+            enumContains<FixedCryptoList>(element.symbol.replace("USDT", ""))
+        }
+    }
+
+    private suspend fun getCryptoPricesList() = binanceApi.get24HourData().body()
+
+    private suspend fun getAllFavourites() = cryptoDao.getAllFavourites()
+
+    private fun Binance24DataItem.toBinanceDataItem(): BinanceDataItem {
+        return BinanceDataItem(
+            symbol = symbol.replace("USDT", ""),
+            last = last,
+            high = high,
+            low = low,
+            open = open,
+            priceChange = priceChange,
+            priceChangePercent = priceChangePercent
+        )
+    }
+
+    private fun List<BinanceDataItem>.applyFavourites(favouriteList: List<FavouriteCrypto>?): List<BinanceDataItem> {
+        return map { item ->
+            item.copy(
+                isFavourite = favouriteList?.any { it.symbol == item.symbol } == true
+            )
+        }
+    }
+
+    //    I will have to start and pause collecting state
+    // It collects message emitted from websocket sharedFlow and updates list item by reassigning BinanceDataItem to new value
+    suspend fun startCollectingBinanceTickerData() {
```

Comment:
Start web socket client values are started here, but are they stopped anywhere? You should create a cancellable job. And add additional method to cancel it like displayed below:
```
private val webSocketJob: Job? = null
suspend fun startCollectingBinanceTickerData() {
    webSocketJob = web_socket_client.state.collect { ... }
}

// method to be added
fun stopCollectingBinanceTickerData() {
    webSocketJob = web_socket_client?.cancel()
}
```